packages/wallet-service/src/api/wallet.ts

```diff
@@ -380,6 +381,65 @@ interface LoadResult {
   xpubkey: string;
 }
 
+/*
+ * This lambda will be started by a SNSMessage on the load failed SNS configured
+ * in serverless.yml. It will receive all wallet load failed events published on
+ * the loadWalletAsync DLQ SNS topic
+ */
+export const loadWalletFailed: Handler<SNSEvent> = async (event) => {
+  const logger = createDefaultLogger();
+  const records = event.Records;
+
+  try {
+    for (let i = 0; i < records.length; i++) {
+      const snsEvent = records[i].Sns;
+      const { RequestID, ErrorMessage } = snsEvent.MessageAttributes;
+
+      // Process each failed load wallet event
+      const loadEvent: LoadEvent = JSON.parse(snsEvent.Message) as unknown as LoadEvent;
+
+      if (!loadEvent.xpubkey) {
+        logger.error('Received wallet load fail message from SNS but no xpubkey received');
+        continue;
+      }
+
+      const walletId = getWalletId(loadEvent.xpubkey);
+
+      // update wallet status to 'error' and set the number of retries to MAX so
+      // it doesn't get retried
+      await updateWalletStatus(mysql, walletId, WalletStatus.ERROR, MAX_LOAD_WALLET_RETRIES);
+
+      logger.error(`${walletId} failed to load.`);
+      logger.error({
+        xpubkey: loadEvent.xpubkey,
```

Comment:
[doubt] Should we log the xpubkey? If this log falls on the hand of malicious actor, could it harm the user in any way?

Reply:
I believe the `walletId` is sufficient to uniquely identify the wallet on the database, right?

Reply:
I don't think the user can be harmed, but if this is leaked, it will indeed be a privacy issue as it will reveal information about the wallet addresses

I agree the `walletId` should be enough, fixed in 9bb99e65b69daf2037241252016cd9a514ba1b07


Thanks